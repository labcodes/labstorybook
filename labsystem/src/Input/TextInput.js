import React from "react";
import PropTypes from "prop-types";

import AbstractTextInput from "./AbstractTextInput";

export default class TextInput extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    theme: PropTypes.string,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    defaultValue: PropTypes.string,
    value: PropTypes.string, // avaliar depois como resolver//
    icon: PropTypes.string,
    iconColor: PropTypes.string,
    required: PropTypes.bool,
    helpMessage: PropTypes.string,
    prefix: PropTypes.string,
    suffix: PropTypes.string,
    isValid: PropTypes.bool,
    customErrorMsg: PropTypes.string,
    onChange: PropTypes.func,
    onIconClick: PropTypes.func,
  };

  static defaultProps = {
    theme: undefined,
    className: undefined,
    disabled: false,
    defaultValue: undefined,
    value: undefined, // avaliar depois como resolver//
    icon: undefined,
    iconColor: "mineral70",
    required: false,
    helpMessage: undefined,
    prefix: undefined,
    suffix: undefined,
    isValid: undefined,
    customErrorMsg: undefined,
    onChange: undefined,
    onIconClick: undefined,
  };

  render() {
    return <AbstractTextInput type="text" {...this.props} />;
  }
}
