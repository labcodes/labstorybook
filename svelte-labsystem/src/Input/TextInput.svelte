<script>
  import AbstractTextInput from "./AbstractTextInput.svelte";

  // we need to replicate the props, just so that svelte logs
  // the warns about missing props using the right component

  export let id;
  export let label;
  export let icon = undefined;
  export let iconColor = undefined;
  export let value = "";

  export let isValid = undefined;
  export let disabled = undefined;
  export let required = undefined;

  export let prefix = undefined;
  export let suffix = undefined;

  export let onIconClick = undefined;
  export let helpMessage = undefined;
  export let customErrorMsg = undefined;
</script>

<AbstractTextInput
  type="text"
  {id}
  {label}
  {icon}
  {iconColor}
  {value}
  {isValid}
  {disabled}
  {required}
  {prefix}
  {suffix}
  {onIconClick}
  {helpMessage}
  {customErrorMsg} />
