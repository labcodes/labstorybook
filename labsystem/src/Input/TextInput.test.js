import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import TextInput from "./TextInput";

describe("TextInput", () => {
  it("renders with base props AbstractTextInput with type text", async () => {
    const renderedComponent = renderer
      .create(<TextInput id="testInput" label="Test Input" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders AbstractTextInput with type=text", async () => {
    const component = mount(<TextInput id="testInput" label="Test Input" />);
    expect(component.find("input[type='text']")).toHaveLength(1);
  });
});
