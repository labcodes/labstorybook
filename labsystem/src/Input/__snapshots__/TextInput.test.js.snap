// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders with base props AbstractTextInput with type text 1`] = `
<div
  className="lab-input "
>
  <input
    autoComplete="off"
    className="lab-input__field "
    id="testInput"
    onChange={[Function]}
    placeholder=" "
    type="text"
    value=""
  />
  <div
    className="lab-input__borders"
  />
  
  
  <div
    className="lab-input__label-wrapper"
  >
    
    <label
      className="lab-input__label"
      htmlFor="testInput"
    >
      Test Input
    </label>
  </div>
  
</div>
`;
